x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1247,20 +1245,55 @@ Collection<Object> createComponents(
 
         cacheInvalidatorRegistry.validate();
 
+        setClosableAndReloadableComponents(components);
+        return components;
+    }
+
+    private void setClosableAndReloadableComponents(List<Object> components) {
+        final List<Object> allComponents = new ArrayList<>(components);
+        // adding additional components we don't expose externally,
+        // but want to allow reloading settings and closing resources
+        // when the security plugin gets closed
+        allComponents.add(remoteClusterSecurityExtension.get());
```

Comment:
Since we only need to iterate over these, you could use 
```
final Iterable<Object> allComponents = Iterables.flatten(
    components, List.of(remoteClusterSecurityExtension.get())
);
```